src/components/SignUpForm.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import { Grid, Button, Form, Input } from "semantic-ui-react";
+import auth from "../modules/auth";
+import { useHistory } from "react-router-dom";
+import "../css/index.css";
+
+const SignUpForm = (props) => {
+  const [message, setMessage] = useState("");
+  const history = useHistory();
+
+  const signup = async (e) => {
+    e.preventDefault();
+    try {
+      let response = await auth.signUp({
+        email: e.target.email.value,
+        password: e.target.password.value,
+        password_confirmation: e.target.passwordConfirmation.value,
+      });
+      if (response.data.success) {
+        props.setUid(response.data.uid);
+        history.push("/sign_in");
```

Comment:
Is this setting the uid as if we were logged in, but it is not setting authenticated after register?
Would it not be better to just set the user as authenticated on success?

Reply:
What happens if you do history.pop() or history.push(history[history.length-2]), do you get redirected to the page you came from? Should we try this?

Reply:
hm, well the user will have a uid set but authenticated will be false. Think that you do need to do the login step.